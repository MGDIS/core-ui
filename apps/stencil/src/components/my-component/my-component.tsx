import { Component, Prop, State, h } from '@stencil/core';
import { format } from '../../utils/utils';

@Component({
  tag: 'my-component',
  styleUrl: 'my-component.css',
  shadow: true,
})
export class MyComponent {
  /**
   * The first name
   */
  @Prop() first: string;

  /**
   * The middle name
   */
  @Prop() middle: string;

  /**
   * The last name
   */
  @Prop() last: string;

  /**
   * counter
   */
  @State() count = 0;

  /**
   * Get formatted text
   * @returns formated text
   */
  private getText(): string {
    return format(this.first, this.middle, this.last);
  }

  /**
   * Add count on click
   */
  private handleClick = () => {
    this.count++;
  };

  render() {
    return (
      <div>
        Hello, World! I'm {this.getText()}
        <mg-button type="button" onClick={this.handleClick} variant="secondary">
          count is {this.count}
        </mg-button>
      </div>
    );
  }
}
