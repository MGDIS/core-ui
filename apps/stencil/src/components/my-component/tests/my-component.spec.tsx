import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MyComponent } from '../my-component';

const getPage = (args?) =>
  newSpecPage({
    components: [MyComponent],
    template: () => <my-component {...args}></my-component>,
  });

describe('my-component', () => {
  it('renders', async () => {
    const { root } = await getPage();
    expect(root).toMatchSnapshot();
  });

  it('renders with values', async () => {
    const { root } = await getPage({ first: 'Stencil', last: "'Don't call me a framework' JS" });
    expect(root).toMatchSnapshot();
  });
});
