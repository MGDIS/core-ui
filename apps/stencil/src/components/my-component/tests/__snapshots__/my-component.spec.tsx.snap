// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my-component renders 1`] = `
<my-component>
  <template shadowrootmode="open">
    <div>
      Hello, World! I'm
      <mg-button type="button" variant="secondary">
        count is 0
      </mg-button>
    </div>
  </template>
</my-component>
`;

exports[`my-component renders with values 1`] = `
<my-component>
  <template shadowrootmode="open">
    <div>
      Hello, World! I'm Stencil 'Don't call me a framework' JS
      <mg-button type="button" variant="secondary">
        count is 0
      </mg-button>
    </div>
  </template>
</my-component>
`;
